recipes/cairo/all/conanfile.py

```diff
@@ -0,0 +1,210 @@
+            del self.options.enable_xcb
+
+    def requirements(self):
+        if self.options.enable_ft:
+            self.requires("freetype/2.10.2")
+        if self.settings.os != "Windows" and self.options.enable_fc:
+            self.requires("fontconfig/2.13.91")
+        if self.settings.os == 'Linux':
+            if self.options.enable_xlib or self.options.enable_xlib_xrender or self.options.enable_xcb:
+                self.requires("xorg/system")
+        if self.options.enable_glib:
+            self.requires("glib/2.65.1")
+        self.requires("zlib/1.2.11")
+        self.requires("pixman/0.40.0")
+        self.requires("libpng/1.6.37")
+
+    def build_requirements(self):
+        if tools.os_info.is_windows:
+            self.build_requires('7zip/19.00')
+            if not tools.get_env("CONAN_BASH_PATH"):
+                self.build_requires('msys2/20190524')
+        self.build_requires("pkgconf/1.7.3")
+        if not self._is_msvc:
+            self.build_requires("libtool/2.4.6")
+
+    @property
+    def _is_msvc(self):
+        return self.settings.compiler == 'Visual Studio'
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+
+        os.rename('cairo-%s' % self.version, self._source_subfolder)
+        
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        if self._is_msvc:
+            self._build_msvc()
+        else:
+            self._build_configure()
+
+    def _build_msvc(self):
+        with tools.chdir(self._source_subfolder):
+            # https://cairographics.org/end_to_end_build_for_win32/
+            win32_common = os.path.join('build', 'Makefile.win32.common')
+            tools.replace_in_file(win32_common, '-MD ', '-%s ' % self.settings.compiler.runtime)
+            tools.replace_in_file(win32_common, '-MDd ', '-%s ' % self.settings.compiler.runtime)
+            tools.replace_in_file(win32_common, '$(ZLIB_PATH)/lib/zlib1.lib',
+                                                self.deps_cpp_info['zlib'].libs[0] + '.lib')
+            tools.replace_in_file(win32_common, '$(LIBPNG_PATH)/lib/libpng16.lib',
+                                                self.deps_cpp_info['libpng'].libs[0] + '.lib')
+            tools.replace_in_file(win32_common, '$(FREETYPE_PATH)/lib/freetype.lib',
+                                                self.deps_cpp_info['freetype'].libs[0] + '.lib')
+            with tools.vcvars(self.settings):
+                env_msvc = VisualStudioBuildEnvironment(self)
+                env_msvc.flags.append('/FS')  # C1041 if multiple CL.EXE write to the same .PDB file, please use /FS
+                with tools.environment_append(env_msvc.vars):
+                    env_build = AutoToolsBuildEnvironment(self)
+                    args=['-f', 'Makefile.win32']
+                    args.append('CFG=%s' % str(self.settings.build_type).lower())
```

Comment:
I think we'll want to limit this to release and debug.
relwithdebinfo and minsizerel are not allowed options, right?